src/main/java/org/isf/opd/gui/OpdEdit.java

```diff
@@ -508,8 +507,7 @@ private JButton getOkButton() {
 
 						try {
 							if (insert) {    // Insert
-								GregorianCalendar date = new GregorianCalendar();
-								opd.setProgYear(opdManager.getProgYear(date.get(Calendar.YEAR)) + 1);
+								opd.setProgYear(opdManager.getProgYear(LocalDate.now().getYear()) + 1);
```

Comment:
In the case OPDEXTENDED=yes we call .getProgYear() method passing the visitDate (so it calculates it within the specified year)

Reply:
Well that is a change from the behavior that I modified here isn't it?   It was using "NOW" and I just did the same.
This is a NEW  (ie, insert) so is there even a visit date?

Reply:
if you like you can blame the forgotten change among OP-719 (my bad)

Reply:
No need to find blame; I've updated the code; please review @mwithi

Reply:
Thanks!